src/lib/dnssd/Discovery_ImplPlatform.cpp

```diff
@@ -53,14 +53,19 @@ static void HandleNodeResolve(void * context, DnssdService * result, CHIP_ERROR
     }
 
     DiscoveredNodeData nodeData;
+
     Platform::CopyString(nodeData.hostName, result->mHostName);
     Platform::CopyString(nodeData.instanceName, result->mName);
 
-    if (result->mAddress.HasValue() && nodeData.numIPs < DiscoveredNodeData::kMaxIPAddresses)
+    if (result->mAddress.HasValue())
+    {
+        nodeData.ipAddress[0] = result->mAddress.Value();
+        nodeData.interfaceId  = result->mInterface;
+        nodeData.numIPs       = 1;
```

Comment:
Just to confirm, are we saying it is OK to always write `ipAddress[0]` because the resolve is already broken for `src/lib/dnssd/Discovery_ImplPlatform.cpp` and always only populates `ipAddress[0]` anyway?

Reply:
In other words, why are we modifying the code that captures IP address in addition to the code that captures interfaceId?

Reply:
I see, alright.

Reply:
The Discovery_ImplPlatform is only processing results one by one. This just makes it clear that this is the behaviour intstead of making it seem that numIPs can increase.

The previous code was likely copied & pasted, hence the extra checks and logic. I find the new code much clearer on what it does.

Reply:
Previous code should have been:

```cpp
nodeData.ipAddress[0]   = result->mAddress.Value();
nodeData.interfaceId[0] = result->mInterface;
nodeData.numIPs = 1;
```

Then my delta would have been a lot cleaner.